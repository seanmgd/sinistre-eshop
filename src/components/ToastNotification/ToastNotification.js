import React, { useEffect, useState } from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { motion } from 'framer-motion'

const ToastNotification = ({ textNotification }) => {
  const [animate, setAnimate] = useState('animated')

  useEffect(() => {
    setTimeout(() => {
      setAnimate('animatedClose')
    }, 3000)
  }, [])

  return (
    <DivContainer variants={variantToast} initial="initial" animate={animate}>
      <Text>{textNotification}</Text>
    </DivContainer>
  )
}

ToastNotification.propTypes = {
  text: PropTypes.string,
}

ToastNotification.defaultProps = {
  text: 'Notificaiton de test',
}

const variantToast = {
  initial: { opacity: 0, y: -10, transition: { duration: 1 } },
  animated: { opacity: 1, y: 0, transition: { duration: 0.5 } },
  animatedClose: { y: 50, opacity: 0, transition: { duration: 1 } },
}

const DivContainer = styled(motion.div)`
  background-color: ${({ theme }) => theme.colors.black.base};
  position: absolute;
  top: 20px;
  right: 0;
  width: 150px;
  overflow: hidden;
`

const Text = styled.p`
  color: ${({ theme }) => theme.colors.grey.light};
  font-size: 12px;
  padding: 8px;
`
export default ToastNotification
