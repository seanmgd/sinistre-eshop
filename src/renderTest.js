import { render as renderTest } from '@testing-library/react'
import { AppProviders } from './AppProviders'
import theme from './styles/theme'

// Export it first to allow overrides.
export * from '@testing-library/react'
export { renderInAppProviders as render }

export function renderInAppProviders(node, options) {
  const { rerender, ...res } = renderTest(
    <AppProviders theme={theme} children={node} />,
    options,
  )

  return {
    ...res,
    rerender: node => rerender(<AppProviders theme={theme} children={node} />),
  }
}
