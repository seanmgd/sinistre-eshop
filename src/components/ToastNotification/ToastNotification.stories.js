import { text } from '@storybook/addon-knobs'
import ToastNotification from './ToastNotification'

export default {
  title: 'Action/ToastNotification',
  component: ToastNotification,
}

export const ToStorybook = () => (
  <ToastNotification
    textNotification={text(
      'Text de la notification',
      ToastNotification.defaultProps.text,
    )}
  />
)

ToStorybook.story = {
  name: 'ToastNotification',
}
