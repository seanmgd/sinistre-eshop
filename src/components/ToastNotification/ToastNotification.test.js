import { render } from '../../renderTest'
import { ToastNotification } from './ToastNotification'

describe('<ToastNotification />', () => {
  it('should render a notification', () => {
    const { container } = render(<ToastNotification />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
