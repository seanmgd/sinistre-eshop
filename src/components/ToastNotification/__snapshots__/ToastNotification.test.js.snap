// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToastNotification /> should render a notification 1`] = `
<div
  class="sc-gsTCUz fQNHPQ"
  style="opacity: 0;"
>
  <p
    class="sc-dlfnbm cXpbAw"
  >
    This is a notification example
  </p>
</div>
`;
